ssf/samples.go

```diff
@@ -1,6 +1,15 @@
 package ssf
 
-import "time"
+import (
+	"math/rand"
+	"time"
+)
+
+// DefaultSampleOptions is a set of SampleOptions that get applied to
+// all samples created in this package, before the more specific
+// sample options get applied. These are convenient for setting metric
+// name prefixes or process-wide sample rates.
+var DefaultSampleOptions []SampleOption
```

Comment:
As I've mentioned, I'm really not a fan of this pattern in Go - it leads to rather convoluted code (see: OpenTracing). If we don't have an immediate alternative ready, I won't block on this, since we have people who want to use the metrics client already, but I'd strongly prefer to use a simpler configuration approach.

Reply:
Yeah, it's true (& the order-dependence of things isn't super great either). I have come to think of this as too general in a way: What we want out of these default sample options is a way to set a metrics prefix; the other options are all just not very useful. Sooooooo I guess I could just make a global `NamePrefix`? That'd probably be easier to document & use!

Reply:
Oh dear, I didn't realize it was order-dependent too.

Taking a step back: the main motivation behind functionally-defined options is that it allows future customization without introducing surface-level breaking changes to the API. (Adding an extra parameter to a function obviously does, but even adding fields to config structs will as well, because config structs *may* be defined with unnamed literals.


However, for Veneur, we're not super-concerned with making surface-level breaking changes that have straightforward migration paths, because we have a predictable release cycle, and the number of callsites others would need to update is pretty small. So I'd bias towards the simpler configuration strategy here for now, even if that means we may change it in a future major release.